e2e/angular-extensions/src/angular-app.test.ts

```diff
@@ -86,3 +86,87 @@ describe('Angular Package', () => {
     }
   });
 });
+
+describe('Angular MFE App Serve', () => {
+  let hostApp;
+  let remoteApp1;
+  let remoteApp2;
+  let proj: string;
+
+  beforeEach(() => {
+    hostApp = uniq('app');
+    remoteApp1 = uniq('remoteApp1');
+    remoteApp2 = uniq('remoteApp2');
+
+    proj = newProject();
+
+    // generate host app
+    runCLI(
+      `generate @nrwl/angular:app ${hostApp} --mfe --mfeType=host --routing --style=css --no-interactive`
+    );
+
+    // generate remote apps
+    runCLI(
+      `generate @nrwl/angular:app ${remoteApp1} --mfe --mfeType=remote --host=${hostApp} --port=4201 --routing --style=css --no-interactive`
+    );
+    runCLI(
+      `generate @nrwl/angular:app ${remoteApp2} --mfe --mfeType=remote --host=${hostApp} --port=4202 --routing --style=css --no-interactive`
+    );
+  });
+
+  afterEach(() => removeProject({ onlyOnCI: true }));
+
+  it('should serve the host and remote apps successfully', () => {
+    // ACT
+    const serveOutput = runCLI(`serve-mfe ${hostApp}`);
+
+    // ASSERT
+    expect(serveOutput).toContain('Compiled successfully');
+    expect(serveOutput).toContain(
+      'Angular Live Development Server is listening on localhost:4200'
+    );
+    expect(serveOutput).toContain(
+      'Angular Live Development Server is listening on localhost:4201'
+    );
+    expect(serveOutput).toContain(
+      'Angular Live Development Server is listening on localhost:4202'
+    );
+  });
+});
+
+describe('Angular App Build and Serve Ops', () => {
+  let app;
+  let proj: string;
+
+  beforeEach(() => {
+    app = uniq('app');
+
+    proj = newProject();
+
+    // generate app
+    runCLI(
+      `generate @nrwl/angular:app ${app} --routing --style=css --no-interactive`
+    );
+  });
+
+  afterEach(() => removeProject({ onlyOnCI: true }));
+
+  it('should build the app successfully', () => {
+    // ACT
+    const serveOutput = runCLI(`build ${app}`);
+
+    // ASSERT
+    expect(serveOutput).toContain('Compiled successfully');
+  });
+
+  it('should serve the app successfully', () => {
+    // ACT
+    const serveOutput = runCLI(`serve ${app}`);
```

Comment:
Please use `runCommandUntil` instead and specify the custom port to avoid potential clashes:

```ts
    const port = 4567;
    const p = await runCommandUntil(
      `serve ${app} --port=${port}`,
      (output) => output.includes(`listening on localhost:${port}`)
    );
    
    // port and process cleanup
    try {
      await promisifiedTreeKill(p.pid, 'SIGKILL');
      expect(await killPorts(port)).toBeTruthy();
    } catch (err) {
      expect(err).toBeFalsy();
    }
```

You can even add `getData` like in `node.test.ts` that checks for content of the served page before we close the process.

Reply:
We use getData to make sure that the page is actually built and not serving an empty/error pages regardless of what compilation says.

Reply:
Changes made